Build/CIRCLECI_SETUP.md

```diff
@@ -0,0 +1,246 @@
+and optionally a GPG key for commit/tag signing.
+
+### UPM branch
+
+Ensure you have a branch called `upm`
+
+### Git SSH key
+
+1. Generate a new SSH key, replacing `your_email@example.com` with
+   the email address of the GitHub account that will be used for the auto
+   deployment.
+
+    ```sh
+    $ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+    ```
+
+    - When you're prompted to "Enter a file in which to save the key,", enter a
+      new location such as `/home/your_user/.ssh/newtonsoft.json-for-unity.converters_rsa`
+
+    - At the prompt, **enter no passphrase**.
+
+2. Copy the content of the public key file
+   (ex: `/home/your_user/.ssh/newtonsoft.json-for-unity.converters_rsa.pub`)
+   into the GitHub project as a deploy key, naming it something like
+   "Newtonsoft.Json-for-Unity.Converters UPM branch deployment key"
+
+    >For jilleJr/Newtonsoft.Json-for-Unity.Converters, that page is found at
+    ><https://github.com/jilleJr/Newtonsoft.Json-for-Unity.Converters/settings/keys/new>
+
+    - Make sure to check the "Allow write access" checkbox.
+
+3. Copy the content of the private key file
+   (ex: `/home/your_user/.ssh/newtonsoft.json-for-unity.converters_rsa`)
+   into the "Add SSH Key" dialog in CircleCI, specifying hostname as
+   "github.com"
+    
+    >For jilleJr/Newtonsoft.Json-for-Unity.Converters, that page is found at
+    ><https://app.circleci.com/settings/project/github/jilleJr/Newtonsoft.Json-for-Unity.Converters/ssh>
+
+4. Copy the fingerprint of the new deploy key shown in CircleCIs UI
+
+5. Update the `deploy-github` job 
+   inside the `.circleci/config.yml` file to use the correct SSH key
+   fingerprint.
+
+    Example, where using the fingerprint
+    `d7:ed:05:51:bd:95:3d:5b:dc:1f:97:00:6a:e5:b0:3c`:
+
+    ```yaml
+      - add_ssh_keys:
+          fingerprints:
+            - "d7:ed:05:51:bd:95:3d:5b:dc:1f:97:00:6a:e5:b0:3c"
+    ```
+
+### Git user info
+
+This is used when creating commits and tags. Just basic user info is needed.
+
+1. Add the git email of user that deploys as variable `GIT_USER_EMAIL`
+   in CircleCI
+
+2. Add the user name (display name) of user that deploys as variable
+   `GIT_USER_NAME` in CircleCI
+
+### Git GPG key
+
+1. Generate key
+   
+    ```sh
+    $ gpg --full-generate-key
+    ```
+
+    Use following generation instructions, **but no passphrase**:
+
+    - kind: (1) RSA and RSA (default)
+    - keysize: 4096
+    - expiration: /up to you, ex: 1y/
+    - real name: /full name of deploying user, first+last name/
+    - email: /git email that's verified on github account of deploying user/
+    - comment: /ex: Json.NET Unity Converters auto deployment to GitHub/
+
+2. Find the newly generated key ID and paste it as variable `GIT_GPG_ID` in
+   CircleCI. This key is referenced as `$KEY_ID` in the steps below.
+
+    ```sh
+    $ gpg --list-keys --keyid-format LONG
+    ```
+
+3. Add GPG public key to your GitHub account <https://github.com/settings/keys>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [[ordered-list-marker-value] Marker should be `1`, was `3`](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=5461394)